src/frontend/js/utils/api/lms/openedx-hawthorn.ts

```diff
@@ -112,8 +112,13 @@ const API = (APIConf: AuthenticationBackend | LMSBackend, options?: APIOptions):
             },
           }),
         })
-          .then((response) => {
+          .then(async (response) => {
             if (response.ok) return response.json();
+            if (response.status === 400) {
+              // Send data to component with localizedMessage
+              const result = await response.json();
```

Comment:
Not sure what happend when the response body is not a json. Should write a test about that.

Reply:
Yeah, in this case I think it's best to add a try catch just to make sure.